schedulearn-frontend/src/components/NestedList.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import ListItem from './ListItem';
+import './NestedList.css';
+import arrow from './back.svg';
+
+export interface Item {
+    index?: number
+    label: string;
+    subItems: Item[];
+}
+
+interface Props {
+    item: Item;
+    onItemClick?(item: Item): void;
+}
+
+export default class NestedList extends React.Component<Props, {}> {
+
+    state: {
+        original?: Item,
+        item?: Item,
+        history: number[],
+    } = {
```

Comment:
All of this can probably be set in the constructor, avoiding making the item nullable, unless there is some specific reason for this